server/app/controllers/applicant/RedirectController.java

```diff
@@ -46,20 +49,20 @@ public RedirectController(
       HttpExecutionContext httpContext,
       ApplicantService applicantService,
       ProfileUtils profileUtils,
-      ProgramRepository programRepository,
+      ProgramService programService,
       ApplicantUpsellCreateAccountView upsellView,
       MessagesApi messagesApi,
       LanguageUtils languageUtils) {
     this.httpContext = checkNotNull(httpContext);
     this.applicantService = checkNotNull(applicantService);
     this.profileUtils = checkNotNull(profileUtils);
-    this.programRepository = checkNotNull(programRepository);
+    this.programService = checkNotNull(programService);
     this.upsellView = checkNotNull(upsellView);
     this.messagesApi = checkNotNull(messagesApi);
     this.languageUtils = checkNotNull(languageUtils);
   }
 
-  public CompletionStage<Result> programByName(Http.Request request, String programName) {
+  public CompletionStage<Result> programByName(Http.Request request, String programSlug) {
```

Comment:
Blocking: Can this be renamed to `programBySlug()`? Otherwise, I'm confused about the variable name change.

Reply:
done